tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,261 @@
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(c.T(), clusterName, "Cluster name to install is not set")
+
+	// Get clusterID with clusterName
+	clusterID, err := clusters.GetClusterIDByName(client, clusterName)
+	require.NoError(c.T(), err)
+
+	// Get latest versions of CIS Benchmark chart
+	latestCisBenchmarkVersion, err := client.Catalog.GetLatestChartVersion(charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	// Create project
+	projectConfig := &management.Project{
+		ClusterID: clusterID,
+		Name:      cisbenchmarkProjectName,
+	}
+	createdProject, err := client.Management.Project.Create(projectConfig)
+	require.NoError(c.T(), err)
+	require.Equal(c.T(), createdProject.Name, cisbenchmarkProjectName)
+	c.project = createdProject
+
+	c.chartInstallOptions = &charts.InstallOptions{
+		ClusterName: clusterName,
+		ClusterID:   clusterID,
+		Version:     latestCisBenchmarkVersion,
+		ProjectID:   createdProject.ID,
+	}
+
+}
+
+func (c *CisBenchmarkTestSuite) TestCisBenchmarkChart() {
+	subSession := c.session.NewSession()
+	defer subSession.Cleanup()
+
+	client, err := c.client.WithSession(subSession)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Installing latest version of cis benchmark chart")
+	err = charts.InstallRancherCisBenchmarkChart(client, c.chartInstallOptions)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting for cisbenchmark chart deployments to have expected number of available replicas")
+	err = charts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	scanName := namegen.AppendRandomString("scan")
+
+	c.T().Log("Running a scan")
+	yamlInput := &management.ImportClusterYamlInput{
+		DefaultNamespace: charts.RancherCisBenchmarkNamespace,
+		YAML:             fmt.Sprintf(ClusterScanYaml, scanName, c.cisProfileName),
+	}
+
+	// get the cluster
+	cluster, err := client.Management.Cluster.ByID(c.project.ClusterID)
+	require.NoError(c.T(), err)
+	// Use ActionImportYaml to the apply the CIS Benchmark yaml file
+	_, err = client.Management.Cluster.ActionImportYaml(cluster, yamlInput)
+	require.NoError(c.T(), err)
+
+	steveClient, err := client.Steve.ProxyDownstream(c.project.ClusterID)
+	require.NoError(c.T(), err)
+
+	scanFailed := false
+
+	err = wait.Poll(1*time.Second, 5*time.Minute, func() (done bool, err error) {
+		// get list of constraints
+		scanList, err := steveClient.SteveType(clusterScanResourceType).List(nil)
+		if err != nil {
+			return false, err
+		}
+
+		scanIdx := -1
+
+		for idx, scan := range scanList.Data {
+			if scan.Name == scanName {
+				scanIdx = idx
+				break
+			}
+		}
+
+		if scanIdx < 0 {
+			return false, errors.New("scan not found")
+		}
+
+		// parse list of constraints
+		clusterScanStatusType := &ClusterScanStatus{}
+		scanStatus := scanList.Data[scanIdx].Status
+		err = v1.ConvertToK8sType(scanStatus, clusterScanStatusType)
+		require.NoError(c.T(), err)
```

Comment:
I don't have a strong opinion, but to prevent mixed error messages because it's in a poll already, this can return false and err, and on line 223, we would see the error there